src/sherpa_ai/agents/base.py

```diff
@@ -119,35 +125,85 @@ def validate_output(self):
+            )
+            for x in range(len(instantiated_validations)):
+                validation = instantiated_validations[x]
+                logger.info(f"validation_running: {validation.__class__.__name__}")
+                logger.info(f"validation_count: {validation.count}")
+                if validation.count < self.validation_steps:
+                    result = self.synthesize_output()
+                    self.belief.update_internal(EventType.result, self.name, result)
+                    validation_result = validation.process_output(
+                        text=result, belief=self.belief
+                    )
+                    logger.info(f"validation_result: {validation_result}")
+                    if not validation_result.is_valid:
+                        self.belief.update_internal(
+                            EventType.feedback,
+                            self.feedback_agent_name,
+                            validation_result.feedback,
+                        )
+                        break
+                    elif x == len(instantiated_validations) - 1:
+                        all_pass = True
+                elif x == len(instantiated_validations) - 1:
+                    validation_is_scaped = True
+                    all_pass = True
+                else:
+                    validation_is_scaped = True
+
+        # if all didn't pass and reached max regeneration run the validation one more time but no regeneration.
+        if validation_is_scaped or self.global_regen_max >= sum(
+            val.count for val in instantiated_validations
+        ):
+            failed_validations = []
 
+            for validation in instantiated_validations:
                 validation_result = validation.process_output(
-                    text=result, belief=self.belief, iteration_count=count
+                    text=result, belief=self.belief
                 )
+                if not validation_result.is_valid:
+                    failed_validations.append(validation)
 
-                if validation_result.is_valid:
-                    result = validation_result.result
-                    break
-                else:
-                    self.belief.update_internal(
-                        EventType.feedback,
-                        self.feedback_agent_name,
-                        validation_result.feedback,
-                    )
-                    result = self.synthesize_output()
-
-            if count >= self.validation_steps:
-                failed_validation.append(validation)
-
-        if len(failed_validation) > 0:
-            # if the validation failed after all steps, append the error messages to the result
             result += "\n".join(
                 failed_validation.get_failure_message()
-                for failed_validation in failed_validation
+                for failed_validation in failed_validations
+            )
+
+        else:
+            result += "\n".join(
```

Comment:
I don't understand the purpose of this branch. Why do we include failure messages from all instantiated validations?